staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go

```diff
@@ -149,18 +150,28 @@ func (o *CreateCronJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, a
 }
 
 func (o *CreateCronJobOptions) Validate() error {
+	if o.Name == "" {
+		return fmt.Errorf("name must be specified")
+	}
+
 	if len(o.Image) == 0 {
 		return fmt.Errorf("--image must be specified")
 	}
+
 	if len(o.Schedule) == 0 {
 		return fmt.Errorf("--schedule must be specified")
 	}
+
 	return nil
 }
 
 func (o *CreateCronJobOptions) Run() error {
 	var cronjob *batchv1beta1.CronJob
 	cronjob = o.createCronJob()
+	if o.EnforceNamespace {
+		cronjob.Namespace = o.Namespace
+		cronjob.Spec.JobTemplate.Namespace = o.Namespace
```

Comment:
Not needed, namespace for the created job will be always inherited from the cronjob itself.